From: Len Brown <len.brown@intel.com>
Subject: Delete ACPI hook from pci_set_power_state()
References: 162320

This small patch removes the ACPI hook in
the PCI power management code.

The intent of that hook was to assure
that PCI devices that also have ACPI
power management methods
would take advantage of those states to
attain the maximum power savings on suspended
laptops.

However, we did not forsee the potential
conflict with ACPI-based PCI-hotplug turning
off slots when subsequent functions on that
slot may still want to access their hardware.

There are other issues with this simple hook
besides hot-plug conflicts.
1. likely ACPI should be called after PCI
   when powering off, and before PCI when
   powering on.
2. devices without PCI-PM are not addressed.

So I think for SLES, we should disable
this hook per this patch; while we try to
come up with something more clever upstream.

Acked-by: <rw@suse.de> 

diff --git a/drivers/pci/pci.c b/drivers/pci/pci.c
index 2329f94..5563d79 100644

--- a/drivers/pci/pci.c
+++ b/drivers/pci/pci.c
@@ -368,13 +368,6 @@ pci_set_power_state(struct pci_dev *dev,
 	else if (state == PCI_D2 || dev->current_state == PCI_D2)
 		udelay(200);
 
-	/*
-	 * Give firmware a chance to be called, such as ACPI _PRx, _PSx
-	 * Firmware method after natice method ?
-	 */
-	if (platform_pci_set_power_state)
-		platform_pci_set_power_state(dev, state);
-
 	dev->current_state = state;
 
 	/* According to section 5.4.1 of the "PCI BUS POWER MANAGEMENT
